paddle/fluid/framework/tensor.cc

```diff
@@ -31,15 +31,16 @@ size_t Tensor::memory_size() const {
   return holder_ == nullptr ? 0UL : holder_->size() - offset_;
 }
 
-void* Tensor::mutable_data(platform::Place place, std::type_index type) {
+void* Tensor::mutable_data(platform::Place place, std::type_index type,
+                           size_t requested_size) {
   if (holder_ != nullptr) {
     holder_->set_type(type);
   }
   PADDLE_ENFORCE_GE(numel(), 0,
                     "When calling this method, the Tensor's numel must be "
                     "equal or larger than zero. "
                     "Please check Tensor::Resize has been called first.");
-  int64_t size = numel() * SizeOfType(type);
+  size_t size = requested_size ? requested_size : numel() * SizeOfType(type);
```

Comment:
@Sand3r- 
- The `request_size` should be larger than `numel() * SizeOfType(type)`?
- Could you add ad PADDLE_ENFORCE to check it in next PR, such as 
```
if (requested_size) 
  PADDLE_ENFORCE_GE(requested_size, numel() * SizeOfType(type));
```

Reply:
Sure thing, I've opened a small PR #12982 for that.